src/replication.c

```diff
@@ -370,23 +386,11 @@ void feedReplicationBuffer(char *s, size_t len) {
             server.repl_backlog->unindexed_count = 0;
         }
     }
-    /* Trim replication backlog. */
+    /* Try to trim replication backlog since replication backlog may exceed
+     * our setting when we add replication stream. */
```

Comment:
```suggestion
     * our setting when we add replication stream. Note that it is important to
     * try to trim at least one node since in the common case this is where one
     * new backlog node is added and one should be removed. see also
     * freeMemoryGetNotCountedMemory. */
```